### ISSUE TITLE
--self insert breaks code in some use cases

### ISSUE THREAD

**Issue by nasht** [nicklockwood/SwiftFormat#123]
Hey Nick,

Thanks for implementing this so quickly.
We found several cases where insert self breaks code.

1. Tupple Var implicit getters.

Old code:

```
 var statusValue: (NSAttributedString, UIColor) {
        var textColor = UIColor.bttDarkGrey
        var lineColor = UIColor.bttMidGrey
   ...

```
After format: `self.UIColor..` is not correct
```
 var statusValue: (NSAttributedString, UIColor) {
        var textColor = self.UIColor.bttDarkGrey
        var lineColor = self.UIColor.bttMidGrey
   ...

```

2. Var declared inside function of same name

Old Code:
```
 fileprivate func commonSuperview(with view: UIView) -> UIView? {
        var commonSuperview: UIView?
        var startView: UIView? = self
        repeat {
            if view.isDescendant(of: startView!) {
                commonSuperview = startView
            }
   ...

```

After format: `self.commonSuperView` is not correct inside the repeat block
```
    fileprivate func commonSuperview(with view: UIView) -> UIView? {
        var commonSuperview: UIView?
        var startView: UIView? = self
        repeat {
            if view.isDescendant(of: startView!) {
                self.commonSuperview = startView
            }
     ...
```

**Comment by nicklockwood** [comment]
@nasht thanks! I've reproduced the first case, but not the second. I think it might depend on some other code in the class. Could you provide some more context?

**Comment by nasht** [comment]
gist shared with you via email

**Comment by nicklockwood** [comment]
Fixed in 0.28.1.
